packages/api/test/e2e/promise-tx.spec.ts

```diff
@@ -192,10 +192,12 @@ describe.skip('Promise e2e transactions', () => {
     return (
       api.rpc.chain.subscribeNewHead(async (header: Header) => {
         if (header.blockNumber.toNumber() === eraDeath - 1) {
-          const hash = await ex.signAndSend(keyring.alice, { blockHash: eraHash, era: exERA, nonce } as any);
-
-          expect(hash).toBeUndefined();
-          done();
+          try {
+            await ex.signAndSend(keyring.alice, { blockHash: eraHash, era: exERA, nonce } as any);
+          } catch (error) {
+            expect(error.message).toMatch(/1010: Invalid Transaction \(0\)/);
+            done();
```

Comment:
So this is a breaking change, or more like a breaking fix.

Before `ex.signAndSend` at an invalid block would resolve with undefined. Now `ex.signAndSend` will throw, which I think should be the desired behavior. cc @KarishmaBothara i think you wrote this test

Reply:
`throw` is correct.